client/components/search-bar.js

```diff
@@ -7,41 +8,39 @@ export default class SearchBar extends Component {
     super(props)
     this.state = {
       isLoading: false,
-      dataBase: [],
       results: []
     }
     this.handleSearch = this.handleSearch.bind(this)
     this.handleBlur = this.handleBlur.bind(this)
-    this.handleClick = this.handleClick.bind(this)
   }
   componentDidMount() {
-    fetch('/artists/')
-      .then(res => res.ok ? res.json() : null)
-      .then(artists => artists && this.setState({ dataBase: artists }))
-      .catch(err => console.error(err))
+    request.sendFetch('/artists/')
+      .then(artists => this.setState({ dataBase: artists }))
   }
   handleSearch(event) {
-    const userSearch = event.target.value
-    const { dataBase } = this.state
-    const results = dataBase.filter(artist => {
-      for (let key in artist) {
-        const artistDetails = artist[key].toString()
-        if (artistDetails.toLowerCase() === userSearch.toLowerCase()) {
-          return true
-        }
+    const debounce = (fn, time) => {
+      let timeout
+
+      return function () {
+        const functionCall = () => fn.apply(this, arguments)
+
+        clearTimeout(timeout)
+        timeout = setTimeout(functionCall, time)
       }
-    })
-    this.setState({ results })
+    }
+    window.addEventListener('keyup', debounce((event) => {
```

Comment:
This is adding a new _additional!!!_ event listener to the window every time you trigger your search function.  Check the Network tab of devTools. Try creating the debounced function in your constructor.

```js
const debounce = (fn, time) => {
  // ...
}
class MyComponent extends Component {
  constructor(props) {
    super(props)
    this.handleSearch = debounce(this.handleSearch.bind(this), 500)
  }
  handleSearch(event) {
    // ...
  }
}
```